src/lib/y2partitioner/dialogs/partition_size.rb

```diff
@@ -38,114 +147,191 @@ def initialize(disk, ptemplate, slots)
           @disk = disk
           @ptemplate = ptemplate
           @slots = slots
+          @largest_region = @slots.map(&:region).max_by(&:size)
+        end
 
-          # FIXME: how much should we support multiple slots?
-          # Maximum size: largest slot
-          # Custom size: smallest slot possible
-          # Custom region: inside any slot
-          @slot = slots.first
-          r = @slot.region
-          # should Region have a #size?
+        def label
+          _("New Partition Size")
+        end
+
+        def items
+          max_size_label = Yast::Builtins.sformat(_("Maximum Size (%1)"),
+            @largest_region.size.human_floor)
+          [
+            [:max_size, max_size_label],
+            [:custom_size, _("Custom Size")],
+            [:custom_region, _("Custom Region")]
+          ]
+        end
 
-          @max_size = r.block_size * r.length
+        def widgets
+          @widgets ||= [
+            MaxSizeDummy.new(@largest_region),
+            CustomSizeInput.new(@slots),
+            CustomRegion.new(@slots)
+          ]
         end
 
         def init
-          Yast::UI.ChangeWidget(Id(:max_size), :Enabled, false)
-          Yast::UI.ChangeWidget(Id(:manual_size), :Enabled, false)
+          self.value = :max_size
+          # trigger disabling the other subwidgets
+          handle("ID" => value)
         end
 
         def store
-          start_block = Yast::UI.QueryWidget(Id(:start_block), :Value)
-          end_block = Yast::UI.QueryWidget(Id(:end_block), :Value)
-          len = end_block - start_block + 1
-          bsize = @slot.region.block_size # where does this come from?
-          region = Y2Storage::Region.create(start_block, len, bsize)
-          @ptemplate.region = region
-        end
-
-        def max_size_term
-          term(
-            :LeftRadioButton,
-            Id(:max_size),
-            Opt(:notify),
-            # radio button text, %1 is replaced by size
-            Builtins.sformat(
-              _("Maximum Size (%1)"),
-              @max_size.to_human_string
-            )
-          )
+          v = value
+          @ids ||= items.map(&:first)
```

Comment:
well, I suggest to have method `ids` and maybe it make sense to have it even in its base class?